pkg/systemd/terminal.jsx

```diff
@@ -95,6 +136,42 @@ const _ = cockpit.gettext;
             document.cookie = cookie;
         }
 
+        onModal(){
+            this.setState({ changePathBusy: false });
+            this.setState(prevState => ({ channel: this.createChannel(prevState.user, cockpit.location.options.path)}));
+        }
+
+        async onNavigate(){
+            // Clear old path errors
+            this.setState({ pathError: null })
+
+            // If there's no path to change to, then we're done here
+            if (!cockpit.location.options.path) {
+                return
+            }
+
+            // Check if path we're changing to exists
+            const exists = await this.dirExists();
+            if (!exists){
+                // Show error and clear location option for path
+                this.setState({ pathError: cockpit.location.options.path })
+                // cockpit.location.replace("")
+            } else if (this.state.pid !== null){
+                // Check if current shell has a process running in it, ie it's busy
+                const cmmd = "grep -qr '^PPid:[[:space:]]*" + this.state.pid + "$' /proc/*/status";
+                cockpit.script(cmmd, [], {err: "message"})
+                    .then(() => {
+                        // it's busy: show a busy modal
```

Comment:
not a modal :grin: But also, the code is clear enough, IMHO doesn't need an extra comment